### DISCUSSION TITLE
ModuleNotFoundError: No module named 'intensity_normalization.type'

### DISCUSSION THREAD

**Discussion by Jkgarciam** [jcreinhold/intensity-normalization#70] [Q&A] [answered]
Hi, I've been unable to import this module, I'm getting this error: 

```python
from intensity_normalization.type import Array, NiftiImage
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'intensity_normalization.type'
```

**Comment by jcreinhold** [comment, marked as answer]
There is no module named `type`. Where are you getting that from? See discussion #62 for more if you're unable to import `intensity_normalization.typing`.

**Reply by Jkgarciam** [reply]
ok, thanks. I got it from here, https://github.com/jcreinhold/intensity-normalization/blob/master/intensity_normalization/normalize/zscore.py

**Reply by jcreinhold** [reply]
I'm still not sure where you found `from intensity_normalization.type import Array, NiftiImage` as no import similar to what you posted appears in the linked to module. Hopefully you found your answer though? There's a `import intensity_normalization.typing as intnormt` which uses the correct name. Perhaps I'm missing context though.

Are you trying to import this package in a script and encountering this error? Or did you write a Python script yourself or open up the interpreter and write: `from intensity_normalization.type import Array, NiftiImage` and see this error message?

If it's the former, you need to run Python 3.9—as I mentioned in discussion #62. Perhaps you're using Python 3.7 or 3.8 or something else which is installing an earlier version of this package; it's hard to tell without more context.

I'd recommend creating a new Python environment with Python version 3.9 and installing the package to see if this resolves your issue. If not, please open an issue and fill out all sections of the issue form so I can help you.

**Comment by Jkgarciam** [comment]
I just noticed that it wasn't the link to the document, anyway I couldn't find it and yes I was doing that on python but since I got some troubles I'm now running everything from the CLI, and it's working well so far, I also wanted to ask you if it's possible to remove outliers of pixels outside a range [-3,3 sigma] for instance.

**Reply by jcreinhold** [reply]
You'll have to do that yourself with your own script, e.g., load the image, extract the image array, and run something like 
```python
sigma = image_array.std()
clipped_image_array = np.clip(image_array, -3.0 * sigma, 3.0 * sigma)
```
and then resave the image if that's what you desire.

**Reply by Jkgarciam** [reply]
thanks!
